src/custom/state/lists/updater.tsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect } from 'react'
+import { unstable_batchedUpdates as batchedUpdates } from 'react-dom'
+import { useDispatch } from 'react-redux'
+
+import { useActiveWeb3React } from '@src/hooks'
+
+import { DEFAULT_ACTIVE_LIST_URLS_BY_NETWORK, DEFAULT_LIST_OF_LISTS_BY_NETWORK } from 'constants/lists'
+import { initialiseTokenLists } from './actions'
+import { ListsState, NEW_LIST_STATE } from './reducer'
+import { buildKey, buildListState } from './helpers'
+
+export default function NetworkListUpdater(): null {
+  const { chainId } = useActiveWeb3React()
+  // const chainIdRef = useRef(chainId)
+
+  const dispatch = useDispatch()
+
+  // Loading//Saving network specific lists from//into storage
+  useEffect(() => {
```

Comment:
All this hook could be simplified. It feels hyper complex and hard to read.  I guess this is why you added a few comments. 

Ideally, it should be easy to read, breaking it down in readable parts. Hopefully, each part is easy to understand what it does. Some example, with pseudocode on how small could be:

```ts
useEffect(() => {
    if (!chainId) {
         return 
    }
    const tokenLists = _getTokenListsFromStorage() state
    
    const  newTokenLists = _getUpdatedListState(networkId, tokenLists)
    dispatch(initialiseTokenLists(newTokenLists))
}
```


Then you continue breaking `_getUpdatedListState` into  readabe stuff, just pseudoce inventing the logic (cause I still don't get it 100%)

```ts
function _getUpdatedListState(networkId, tokenLists) {
       if (!tokenLists) {
           // initialize state
           return buildListState(...)
       } 
        
       mergeXXXX
       doThis(...)
       tokenLists = doThat(...)
      return tokenLists
}
```

Reply:
again this is mainly copy logic from the reducer. I agree it doesn't read particularly well, hence the comments, but i kept it this way to compare later should they change logic and i need to check against the original. Otherwise I'm happy to do what I wanted from the beginning and write it as you are proposing :shrug:

Reply:
Sorry, is just my head does my head 🤯 when I try to understand the updater. 
I get from your comment, that is “the original" code. But I don't find it and it compares line by line. Where does it come from exactly?